internal/vault/vault.go

```diff
@@ -201,6 +209,29 @@ func (v *Vault) newConfig() (*vault.Config, error) {
 	return cfg, nil
 }
 
+func (v *Vault) caBundleFromSecret() ([]byte, error) {
+	ref := v.issuer.GetSpec().Vault.CABundleSecretRef
+
+	var namespace string
+	if ref.Namespace != "" {
+		namespace = ref.Namespace
+	} else {
+		namespace = v.namespace
+	}
```

Comment:
This will be the issuer Namespace or the cluster resource namespace in the case of ClusterIssuer.

Reply:
We need to add a `issuerForConfigMap` [here](https://github.com/cert-manager/cert-manager/blob/6947696b12428a1f9e5823c2e69de84b8dd08c1c/pkg/controller/issuers/checks.go#L27), which will re-sync the issuer and [ClusterIssuer](https://github.com/cert-manager/cert-manager/blob/master/pkg/controller/clusterissuers/checks.go#L27) if a ConfigMap is reconciled for the CA Bundle. 

We will need to add support for informing on ConfigMaps for both controllers, like we do for [Secrets](https://github.com/cert-manager/cert-manager/blob/6947696b12428a1f9e5823c2e69de84b8dd08c1c/pkg/controller/clusterissuers/controller.go#L92).

Reply:
Please add appropriate tests to [`vault_test.go`](https://github.com/cert-manager/cert-manager/blob/6947696b12428a1f9e5823c2e69de84b8dd08c1c/internal/vault/vault_test.go#L884).